zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/ProcessInstanceQueryControllerTest.java

```diff
@@ -534,4 +536,42 @@ void shouldSearchProcessInstancesWithAdvancedFilter(
     verify(processInstanceServices)
         .search(new ProcessInstanceQuery.Builder().filter(filter).build());
   }
+
+  @Test
+  void shouldSearchProcessInstancesByState() {
```

Comment:
🔧 I believe we can make this a parameterized test, using the enum class. You can remove the loop from the test method then.

```suggestion
  @ParameterizedTest
  @EnumSource(ProcessInstanceStateEnum.class) 
  void shouldSearchProcessInstancesByState(ProcessInstanceStateEnum state) {
```